rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/search/issue9606.yml

```diff
@@ -18,7 +18,7 @@ setup:
 "Test search_type=query_and_fetch not supported from REST layer":
 
   - do:
-      catch: bad_request
+      catch: /No search type for \[query_and_fetch\]/
```

Comment:
This doesn't necessarily need to change, we still throw the same kind of exception. I was wondering if it makes sense to delete the entire test since it seems a bit obsolete, wdyt?

Reply:
I removed this and added a somewhat similar unit test to check we throw exceptions on unknown values